keras/applications/xception.py

```diff
@@ -264,6 +257,9 @@ def Xception(include_top=True, weights='imagenet',
     elif weights is not None:
         model.load_weights(weights)
 
+    if K.backend() == 'theano':
+        layer_utils.convert_all_kernels_in_model(model)
```

Comment:
To follow the pattern of other applications, please put this conditional inside the `if weights == 'imagenet'` block

Reply:
@fchollet, Updated.